src/styles/utils/__functions.scss

```
@@ -0,0 +1,29 @@
+@use "../utils/_variables" as *;
+
+@for $date from 1 through 31 {
+  .calendar__cell--#{$date}::before {
+    content: "#{$date}";
+  }
+}
+
+@for $day from 28 through 31 {
+  .calendar--month-length--#{$day}
+  :nth-child(n + #{$day + 1}) {
+    display: none;
+  }
+}
+
+$start-day:
+  "mon" 0,
+  "tue" calc(#{$cell-size} + #{$cell-gap}),
+  "wed" calc(2 * (#{$cell-size} + #{$cell-gap})),
```

You can only calculate your margin once, instead of calculating it every time

```suggestion
  "tue" $day-margin,
  "wed" 2 * $day-margin,
```